src/nvim/ui_bridge.c

```diff
@@ -305,6 +306,18 @@ static void ui_bridge_update_bg_event(void **argv)
   ui->update_bg(ui, PTR2INT(argv[1]));
 }
 
+static void ui_bridge_update_sp(UI *b, int sp)
+{
+  UI_CALL(b, update_sp, 2, b, INT2PTR(sp));
+}
+static void ui_bridge_update_sp_event(void **argv)
+{
+  UI *ui = UI(argv[0]);
+  if (ui->update_sp) {
```

Comment:
This check is here since the tui doesn't use the special color field.

An alternative could be just adding a stub to the tui's UI struct.